src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1921,6 +1921,8 @@
     <Field Name="CollectionCreation" Type="BoundExpression?" Null="allow" SkipInVisitor="true"/>
     <!-- Placeholder for collection builder elements span. -->
     <Field Name="CollectionBuilderSpanPlaceholder" Type="BoundValuePlaceholder?" SkipInVisitor="true" Null="allow"/>
+    <!-- Indexer setter, if any. -->
+    <Field Name="IndexerSetMethod" Type="MethodSymbol?" Null="allow"/>
```

Comment:
nit: would it be worthwhile to add a `Validate` method to this bound node to enforce that a `setMethod` is provided when the collectionTypeKind correspond to a dictionary type?

Reply:
I'll avoid creating a `Validate` for now if `IndexerSetMethod` is the only value to check, since we already require this to be set for lowering.